lib/packwerk/commands/update_deprecations_command.rb

```diff
@@ -0,0 +1,82 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "sorbet-runtime"
+require "benchmark"
+
+require "packwerk/commands/offense_progress_marker"
+require "packwerk/commands/result"
+require "packwerk/run_context"
+require "packwerk/updating_deprecated_references"
+
+module Packwerk
+  module Commands
+    class UpdateDeprecationsCommand
+      extend T::Sig
+      include OffenseProgressMarker
+      sig do
+        params(
+          files: T::Enumerable[String],
+          configuration: Configuration,
+          offenses_formatter: Formatters::OffensesFormatter,
+          progress_formatter: Formatters::ProgressFormatter
+        ).void
+      end
+      def initialize(files:, configuration:, offenses_formatter:, progress_formatter:)
+        @files = files
+        @configuration = configuration
+        @progress_formatter = progress_formatter
+        @offenses_formatter = offenses_formatter
+      end
+
+      sig { returns(Result) }
+      def run
+        @progress_formatter.started(@files)
+
+        all_offenses = T.let([], T.untyped)
+        execution_time = Benchmark.realtime do
+          all_offenses = @files.flat_map do |path|
+            run_context.process_file(file: path).tap do |offenses|
+              mark_progress(offenses: offenses, progress_formatter: @progress_formatter)
+            end
+          end
+
+          updating_deprecated_references.dump_deprecated_references_files
+        end
+
+        @progress_formatter.finished(execution_time)
+        calculate_result(all_offenses)
+      end
+
+      private
+
+      sig { returns RunContext }
+      def run_context
+        @run_context = T.let(@run_context, T.nilable(RunContext))
+        @run_context ||= RunContext.from_configuration(
+          @configuration,
+          reference_lister: updating_deprecated_references
+        )
+      end
+
+      sig { returns UpdatingDeprecatedReferences }
+      def updating_deprecated_references
+        @updating_deprecated_references = T.let(
```

Comment:
This change was to prevent sorbet complaining about memoization, but now when running tests I get

`lib/packwerk/commands/update_deprecations_command.rb:65: warning: instance variable @updating_deprecated_references not initialized`

any ideas?

Reply:
Unfortunately I have no idea. Have you tried to define the instance variable in the initializer?